crates/ruff_python_parser/src/lib.rs

```diff
@@ -417,6 +417,15 @@ impl Tokens {
         };
         &self[start..=start + end]
     }
+
```

Comment:
I'm not sure in which PR you added `tokens_in_range` and I also don't know if it is a good idea. 

We could avoid the `end` binary search by either:
* doing a linear serach from the start. That would be based on the assumption that the end is close to the start (linear search has better cache locality)
* Return a custom iterator that lazily checks `end` in the `next` call. I don't know if that's feasible or if any logic depends on the fact that `tokens_in_range` returns a slice.

Reply:
> I also don't know if it is a good idea.

Why though?

A lot of usages of the lexer are to get the tokens within a specified range, usually the range belongs to a node. This does mean that the token tree would be more useful than this.

> We could avoid the `end` binary search by either:
> 
> * doing a linear serach from the start. That would be based on the assumption that the end is close to the start (linear search has better cache locality)
> * Return a custom iterator that lazily checks `end` in the `next` call. I don't know if that's feasible or if any logic depends on the fact that `tokens_in_range` returns a slice.

I can explore this ideas once the code compiles.

Reply:
> Why though?

Sorry, I phrased this poorly. The `tokens_in_range` is a good idea. I'm not sure if my proposal of not using a binary search is a good idea.